### PR TITLE
Intelligence

### PR DESCRIPTION
**Pull request by lnchambers**
Description of Changes:
-
* Made spec harness pass a bunch more
* Fixed minor problems
*

### Tests Added:
* None
*

#### Reference Related Issue(s) or Spec Requirement:
- #.

#### Checklist:
- [x] Code follows code style of this project
- [x] Tests added to cover changes
- [x] All new and existing tests passing
#

#### Questions / Additional Notes:

@mgmilton @lnchambers

### CODE REVIEW

File: app/controllers/api/v1/invoice_items/invoice_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Api::V1::InvoiceItems::InvoiceItemsController < ApplicationController
```
Comment (mgmilton) [inline app/controllers/api/v1/invoice_items/invoice_items_controller.rb:1]: like the creation of these extra controllers. elegant.

File: app/models/invoice_item.rb [hunk 2]
```diff
@@ -2,4 +2,6 @@ class InvoiceItem < ApplicationRecord
   validates_presence_of :unit_price, :quantity
   belongs_to :item
   belongs_to :invoice
+
+  default_scope { order(:id) }
```
Comment (mgmilton) [inline app/models/invoice_item.rb:6]: great scope addition.

File: app/models/merchant.rb [hunk 3]
```diff
@@ -12,4 +13,8 @@ def self.most_revenue(limit = 5)
       .order("revenue DESC").limit(limit)
       .group(:id)
   end
+
+  def favorite_customer
```
Comment (mgmilton) [inline app/models/merchant.rb]: remove this favorite_customer method.

File: app/models/merchant_search.rb [hunk 4]
```diff
@@ -22,9 +18,13 @@ def single_search(params)
     elsif params[:unit_price]
       Merchant.find_by("unit_price LIKE ?", params[:unit_price])
     elsif params[:created_at]
-      Merchant.find_by("created_at LIKE ?", params[:created_at])
+      Merchant.find_by(created_at: Time.zone.parse(params[:created_at]))
```
Comment (mgmilton) [inline app/models/merchant_search.rb:21]: great addition of filtering the time here!

File: app/models/merchant.rb (hunk 3 above)
Comment (lnchambers) [reply app/models/merchant.rb]: It has been removed

File: app/models/search.rb [hunk 5]
```diff
@@ -37,8 +37,12 @@ def single_search(params, model)
       model.find_by("result LIKE ?", params[:result])
     elsif params[:credit_card_number]
       model.find_by("credit_card_number LIKE ?", params[:credit_card_number])
+    elsif params[:updated_at]
+      model.find_by(updated_at: Time.zone.parse(params[:updated_at]))
+    elsif params[:id]
+      model.find_by(id: params[:id])
```
Comment (mgmilton) [inline app/models/search.rb:43]: this is one beastly method. after we finish everything today, let's give refactoring it a shot?

Comment (lnchambers) [reply app/models/search.rb:43]: Of course

File: app/models/search.rb [hunk 6]
```diff
@@ -66,13 +74,25 @@ def multiple_search(params, model)
...
-      model.where("updated_at LIKE ?", params[:updated_at])
+      model.all
     end
   end
 
   def random(model)
     model.all.sample
   end
 
+  def price_search_format(params)
```
Comment (mgmilton) [inline app/models/search.rb:94]: this is a really nice feature.

**Review by mgmilton — APPROVED** [review]
Very impressive work, Luke. Great job coordinating our system with the requirements of the spec harness.

**Pull request merged**